components/tidb_query_codegen/src/rpn_function.rs

```diff
@@ -519,14 +531,43 @@ impl ValidatorFnGenerator {
 }
 
 fn generate_init_metadata_fn(
-    metadata_ctor: &Option<TokenStream>,
+    metadata_type: &Option<TokenStream>,
+    metadata_mapper: &Option<TokenStream>,
     impl_generics: &ImplGenerics<'_>,
     where_clause: Option<&WhereClause>,
 ) -> TokenStream {
-    let fn_body = if let Some(metadata_ctor) = metadata_ctor {
-        quote! { #metadata_ctor(expr).map(|metadata| Box::new(metadata) as Box<(dyn std::any::Any + std::marker::Send + 'static)>) }
-    } else {
-        quote! { Ok(Box::new(())) }
+    let parse_body_gen = |metadata_type: &TokenStream| {
+        quote! {
+            (if expr.get_val().len() == 0 {
```

Comment:
How about extracting this logic into `rpn_expr/types/function.rs` (like `validate_expr_arguments_eq`), so that you can change it to a simple function call?